pkg/cnbutils/copy_project.go

```diff
@@ -11,23 +12,94 @@ import (
 	ignore "github.com/sabhiram/go-gitignore"
 )
 
-func CopyProject(source, target string, include, exclude *ignore.GitIgnore, utils BuildUtils) error {
-	sourceFiles, _ := utils.Glob(path.Join(source, "**"))
+func isFiltered(path string, knownSymlinks []string) bool {
+	for _, symlink := range knownSymlinks {
+		if strings.HasPrefix(path, symlink) {
+			return true
+		}
+	}
+	return false
+}
+
+func filterSourceFiles(sourceFiles []string, utils BuildUtils) ([]string, error) {
+	filteredFiles := []string{}
+	knownSymlinks := []string{}
+
+	for _, sourceFile := range sourceFiles {
+		if isFiltered(sourceFile, knownSymlinks) {
+			continue
+		}
+
+		isSymlink, err := symlinkExists(sourceFile, utils)
+		if err != nil {
+			return nil, err
+		}
+
+		isDir, err := utils.DirExists(sourceFile)
+		if err != nil {
+			return nil, err
+		}
+
+		if isSymlink {
+			if isDir {
```

Comment:
Do we even need this `if`? Apart of the log message have to be rephrased, why not add files to `knownLinks`? It is not needed, but there shouldn't be a conflict when checking `strings.hasPrefix`.